cypress/support/step_definitions/testSetup.js

```diff
@@ -55,6 +55,21 @@ Given('the environment has been set up for {string} tests', (test) => {
     if (Cypress.env(CONSTANTS.TEST_TYPE).includes('rpc-Only')) {
       Cypress.env(CONSTANTS.IS_RPC_ONLY, true);
     }
+    // fetch device details dynamically
+    if (Cypress.env(CONSTANTS.TEST_TYPE) == CONSTANTS.DEVICE_ENV) {
+      cy.getDeviceData(CONSTANTS.DEVICE_ID, {}, CONSTANTS.ACTION_CORE.toLowerCase()).then(
+        (response) => {
+          if (response) {
+            const method = CONSTANTS.REQUEST_OVERRIDE_CALLS.FETCHDEVICEDETAILS;
+            const requestMap = {
+              method: method,
+              params: response,
+            };
+            cy.sendMessagetoPlatforms(requestMap);
+          }
+        }
+      );
+    }
```

Comment:
where is the flag from config module which would skip fetching data for config module dynamically

Reply:
wouldn't this fail in other platforms

Reply:
added an env value in config.json to determine whether dynamic value fetch is supported